File: company-service/src/main/java/ru/panic/companyservice/configuration/CorsConfiguration.java
package ru.panic.companyservice.configuration;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@RestController
@RequestMapping("/api/company")
public class CorsConfiguration implements WebMvcConfigurer {
    @Override
    public void addCorsMappings(CorsRegistry corsRegistry) {
        corsRegistry.addMapping("/api/company/**")
                .allowedHeaders("*")
                .allowedMethods("POST")
                .allowedOrigins("http://localhost:3000");
    }
}

File: company-service/src/main/kotlin/ru/panic/companyservice/configuration/CorsConfiguration.kt
package ru.panic.companyservice.configuration

import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.CorsRegistry
import org.springframework.web.servlet.config.annotation.EnableWebMvc
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
@EnableWebMvc
class CorsConfiguration : WebMvcConfigurer {
    override fun addCorsMappings(corsRegistry: CorsRegistry) {
        corsRegistry.addMapping("/api/company/**")
                .allowedHeaders("*")
                .allowedMethods("POST")
                .allowedOrigins("http://localhost:3000")
    }
}